src/Http/Event/UpdateAttendanceModeRequestHandler.php

```diff
@@ -0,0 +1,94 @@
+use CultuurNet\UDB3\Http\Request\Body\JsonSchemaValidatingRequestBodyParser;
+use CultuurNet\UDB3\Http\Request\Body\RequestBodyParserFactory;
+use CultuurNet\UDB3\Http\Request\RouteParameters;
+use CultuurNet\UDB3\Http\Response\NoContentResponse;
+use CultuurNet\UDB3\Model\ValueObject\Virtual\AttendanceMode;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+final class UpdateAttendanceModeRequestHandler implements RequestHandlerInterface
+{
+    private CommandBus $commandBus;
+
+    private RepositoryInterface $eventRelationsRepository;
+
+    public function __construct(CommandBus $commandBus, RepositoryInterface $eventRelationsRepository)
+    {
+        $this->commandBus = $commandBus;
+        $this->eventRelationsRepository = $eventRelationsRepository;
+    }
+
+    public function handle(ServerRequestInterface $request): ResponseInterface
+    {
+        $routeParameters = new RouteParameters($request);
+        $eventId = $routeParameters->getEventId();
+
+        $parser = RequestBodyParserFactory::createBaseParser(
+            new JsonSchemaValidatingRequestBodyParser(JsonSchemaLocator::EVENT_ATTENDANCE_MODE_PUT),
+            new DenormalizingRequestBodyParser(
+                new AttendanceModeWithLocationDenormalizer(),
+                AttendanceModeWithLocation::class
+            )
+        );
+
+        /** @var AttendanceModeWithLocation $attendanceModeWithLocation */
+        $attendanceModeWithLocation = $parser->parse($request)->getParsedBody();
+
+        $this->guardLocation($eventId, $this->eventRelationsRepository, $attendanceModeWithLocation);
+
+        $commands = [
+            new UpdateAttendanceMode(
+                $eventId,
+                $attendanceModeWithLocation->getAttendanceMode()
+            ),
+        ];
+
+        if ($attendanceModeWithLocation->getLocationId()) {
+            $commands[] = new UpdateLocation(
+                $eventId,
+                $attendanceModeWithLocation->getLocationId()
+            );
+        }
+
+        foreach ($commands as $command) {
+            $this->commandBus->dispatch($command);
+        }
+
+        return new NoContentResponse();
+    }
+
+    private function guardLocation(
+        string $eventId,
+        RepositoryInterface $eventRelationsRepository,
+        AttendanceModeWithLocation $attendanceModeWithLocation
+    ): void {
+        if ($attendanceModeWithLocation->getLocationId() === null &&
+            !$attendanceModeWithLocation->getAttendanceMode()->sameAs(AttendanceMode::online())) {
+            $location = $eventRelationsRepository->getPlaceOfEvent($eventId);
+
+            if ($location === null || $location === LocationId::VIRTUAL_LOCATION) {
+                throw ApiProblem::bodyInvalidData(
+                    new SchemaError('/location', 'The event has no related location, please provide a location')
```

Comment:
Small nit, but:

```suggestion
                    new SchemaError('/location', 'A location is required when changing an online event to mixed or offline')
```

might be a bit clearer?